### PR TITLE
Delete expired locks (older than 1 day)

### PR DESCRIPTION
**Pull request by hmusum**
* There has never been any cleanup of locks, but until we started taking
  a lock per session this has not been seen as an issue, since there were
  few of them
* Add code to maintainer that deletes locks older than 1 day

Due to the way locks are implemented and (they are removed if there is just one process that takes the lock and releases it) there are no test added for this change.  Tested in a config server cluster in CD.

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/maintenance/SessionsMaintainer.java [hunk 1]
```diff
@@ -35,5 +35,9 @@ protected void maintain() {
             int deleted = applicationRepository.deleteExpiredRemoteSessions(expiryTime);
             log.log(LogLevel.FINE, "Deleted " + deleted + " expired remote sessions, expiry time " + expiryTime);
         }
+
+        Duration lockExpiryTime = Duration.ofDays(1);
+        int deleted = applicationRepository.deleteExpiredLocks(lockExpiryTime);
+        log.log(LogLevel.INFO, "Deleted " + deleted + " expired locks, expiry time " + lockExpiryTime);
```
Comment (gjoranv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/maintenance/SessionsMaintainer.java:41]: nit: the middle part of the message could be simplified to " locks older than "

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 2]
```diff
@@ -665,7 +687,7 @@ public Lock lock(long sessionId) {
         return curator.lock(lockPath(sessionId), Duration.ofMinutes(1)); // These locks shouldn't be held for very long.
     }
 
-    private Path lockPath(long sessionId) {
+    public Path lockPath(long sessionId) {
```
Comment (gjoranv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:690]: intended? I don't see any new usage in this PR.

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 3]
```diff
@@ -281,6 +282,27 @@ public int deleteExpiredRemoteSessions(Clock clock, Duration expiryTime) {
...
+            if (zooKeeperNodeCreated(path).isBefore(clock.instant().minus(expiryTime))) {
+                log.log(Level.INFO, "Lock  " + path + " has expired, deleting it");
+                curator.delete(path);
+                deleted++;
+            }
+        }
+        return deleted;
+    }
+
+    private Instant zooKeeperNodeCreated(Path path) {
```
Comment (gjoranv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:298]: perhaps this should return an Optional instead, and empty where it now returns the current time?

**Review by gjoranv — APPROVED** [review]

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java (hunk 2 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:690]: I tried to write a test that needed this (but that was abandoned), so not anymore, will fix

File: configserver/src/main/java/com/yahoo/vespa/config/server/maintenance/SessionsMaintainer.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/maintenance/SessionsMaintainer.java:41]: thanks, will fix

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java (hunk 3 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:298]: Yes, probably better, thanks

**Comment by hmusum** [comment]
@gjoranv please take a look again

**Review by gjoranv — COMMENTED** [review]
👍

**Pull request merged**